azure/spoke/variables.tf

```diff
@@ -22,60 +22,64 @@ variable "resource_group_name" {
 # Network
 ##########
 
-variable "network_security_group_name" {
-  description = "The name of the network security group."
-  type        = string
-  default     = null
-}
-
-variable "route_table_name" {
-  description = "The name of the route table."
+variable "virtual_network_name" {
+  description = "The name of the spoke virtual network."
   type        = string
-  default     = null
-}
-
-variable "disable_bgp_route_propagation" {
-  description = "Boolean flag which controls propagation of routes learned by BGP on that route table. True means disable."
-  type        = bool
-  default     = true
 }
 
-variable "default_route_ip" {
-  description = "The IP address to use for the default route if creating a route table."
-  type        = string
-  default     = null
+variable "address_space" {
+  description = "The address spaces of the virtual network."
+  type        = list(string)
 
   validation {
-    condition     = can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", var.default_route_ip))
-    error_message = "Invalid IP address provided."
+    error_message = "Must be valid IPv4 CIDR."
+    condition     = can(cidrhost(one(var.address_space[*]), 0))
```

Comment:
Your address space variable should not be a list if your validation only allows for 1 item to exist in the list. We should change the variable a string if we're only going to support (by validation) a single item in the list.

Reply:
This will be addressed in issue https://github.com/ans-coe/terraform-azurerm-virtual-network/issues/9